### PR TITLE
Introduce the TouchScrollable component

### PR DESCRIPTION
**Pull request by jossmac**
Simplify consumer implementation by removing the necessity for a `touchScrollTarget` prop that passes refs around. Used like:

```jsx
import ScrollLock, { TouchScrollable } from 'react-scrolllock';

// lock children will be wrapped by TouchScrollable

<ScrollLock>
  <div style={{ height: 200, overflowY: 'auto' }}>{/* ... */}</div>
</ScrollLock>

// or, if you're app structure doesn't allow the above

<ScrollLock />
<TouchScrollable>
  <div style={{ height: 200, overflowY: 'auto' }}>{/* ... */}</div>
</TouchScrollable>
```

Also introduces a companion component, which is a thin wrapper around ScrollLock:

```jsx
<ScrollLockToggle isActive={someProp} />
```

### CODE REVIEW

**Comment by joeruello** [comment]
We just ran into #33 as well, anything we can do to help get this merged?

**Comment by theKashey** [comment]
So - when?

File: src/withTouchListeners.js [hunk 1]
```diff
@@ -15,15 +20,19 @@ export default function withTouchListeners(WrappedComponent: ComponentType<*>) {
...
+      if (!canUseEventListeners) return;
 
       const { touchScrollTarget } = this.props;
       const target = document.body;
 
       // account for touch devices
       if (target && isTouchDevice()) {
         // Mobile Safari ignores { overflow: hidden } declaration on the body.
-        target.addEventListener('touchmove', preventTouchMove, this.listenerOptions);
+        target.addEventListener(
```
Comment (theKashey) [inline src/withTouchListeners.js]: If touchScrollTarget is not provided this logic should be skipped.

**Comment by jossmac** [comment]
`4.0.0` is available on npm

File: src/TouchScrollable.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      listenerOptions,
+    );
+    this.scrollableArea.removeEventListener(
+      'touchmove',
+      allowTouchMove,
+      listenerOptions,
+    );
+  }
+  render() {
+    return <NodeResolver innerRef={this.getScrollableArea} {...this.props} />;
```
Comment (theKashey) [inline src/TouchScrollable.js]: That would not work with portals.

**Pull request merged**